### PR TITLE
try skip unless optimization

### PR DESCRIPTION
**Pull request by isabelrios**
Just trying to get the optimization working to build focus or firefox depending on the folders / files changed

### CODE REVIEW

**Comment by isabelrios** [comment]
@ahal I opened a PR here to debug... I don't get the `build-firefox-debug` task running when I add the optimization.

I tried locally to run `taskgraph optimized` but I get an error (with and without the optimization added: RuntimeError: Could not detect Taskcluster instance, set $TASKCLUSTER_ROOT_URL)

I used `taskgraph full -J --tasks build` and  I can see that the optimization is printed... 
{
  "build-build-firefox-debug": {
    "attributes": {
      "always_target": false,
      "kind": "build",
      "run_on_projects": [
        "all"
      ],
      "run_on_tasks_for": [
        "github-pull-request",
        "github-push"
      ],
      "shipping_phase": null
    },
    "dependencies": {},
    "description": "Build firefox_configure_build bitrise workflow",
    "if_dependencies": [],
    "kind": "build",
    "label": "build-build-firefox-debug",
    **"optimization": {
      "skip-unless-changed": [
        "bitrise.yml"
      ]**
    },

Not sure where to look at to debug this... any clue will be helpful, thanks!

**Comment by ahal** [comment]
Thanks I'll dig into it now

Re: the error, run `export TASKCLUSTER_ROOT_URL=https://firefox-ci-tc.services.mozilla.com/` (I recommend putting this in a bashrc file or equivalent)

**Comment by ahal** [comment]
Narrowing it down.. If you open up the [Decision task artifacts](https://firefox-ci-tc.services.mozilla.com/tasks/K1Jjdpm9RRmZYUMPZT2Z1w#artifacts) and download the `parameters.yml` one, you'll see that it has `files_changed: []`.. So that's why the task is being optimized.

I'm not sure why it thinks there were no changed files... I'm suspecting I regressed this somehow with:
https://github.com/taskcluster/taskgraph/commit/bfa5fe93bf160a97df9ac3d99482765af1f273a8

**Comment by ahal** [comment]
Yes, I found the regression.. This will need a new Taskgraph release, I'll get it fixed asap. Thanks for finding this!

**Comment by isabelrios** [comment]
> Yes, I found the regression.. This will need a new Taskgraph release, I'll get it fixed asap. Thanks for finding this!

Oh nice coincidence to find an issue :) Thanks! There is no rush, we can work on other stuff while that is fixed.

**Comment by isabelrios** [comment]
@ahal I have synced with firefox repo to get the patch.. At least that's working, we see the files changed in that [parameters.yml](https://firefox-ci-tc.services.mozilla.com/tasks/Q6LsR1tAQ-yZUUlcsDrVIQ#artifacts) file:
```
files_changed:
- bitrise.yml
- taskcluster/kinds/build/kind.yml
```

However, the build-firefox-debug task is not triggered yet ... do we need that optimization code in main so that it works? I can land that and check again...

File: taskcluster/kinds/build/kind.yml [hunk 1]
```diff
@@ -7,9 +7,24 @@ loader: taskgraph.loader.transform:loader
...
     - ffios_taskgraph.transforms.bitrise
     - taskgraph.transforms.task
-
+task-defaults:
+    worker-type: bitrise
+    bitrise:
+        artifact_prefix: public
 
 tasks:
+    build-firefox-debug:
```
Comment (ahal) [inline taskcluster/kinds/build/kind.yml:16]: nit: the `kind` implicitly gets added to the name here, so the task label will actually be `build-build-firefox-debug`. You can remove the `build` here to fix it.

**Comment by isabelrios** [comment]
Closing this PR for now. We know already how to work with the optimization. Thanks!

**Pull request closed without merging**